crates/rover-client/src/error.rs

```diff
@@ -163,6 +163,20 @@ pub enum RoverClientError {
 
     #[error("This endpoint doesn't support subgraph introspection via the Query._service field")]
     SubgraphIntrospectionNotAvailable,
+
+    #[error("The input provided is invalid")]
+    InvalidInputError { msg: String },
+
+    #[error("You don't have the required permissions to perform this operation")]
+    PermissionError { msg: String },
+
+    #[error(
+        "You cannot perform this operation due to a limit imposed by your current billing plan"
+    )]
+    PlanError { msg: String },
+
+    #[error("Your check took too long to run")]
```

Comment:
can we add a link to the running check here so folks can head on over to studio to keep watching?

Reply:
It's not possible to link to a pending check because the checks page url is only created after the check starts. This is a problem we're hoping to address in the future by creating a "workflow" level checks page in studio.

Reply:
there's gotta be _some_ page we can link to though, no? even just the top level checks page in studio would be fine to link to here. I just don't want someone to encounter this error, see "your check took too long to run" and for them to be frustrated and think it didn't work when in reality, it's still running and they can go check in on it.